### PR TITLE
Updated the Pydantic functions used by the API

### PR DESCRIPTION
**Pull request by Igor8mr**
Updated the Pydantic functions used by the API, removing the json(sort_keys=True) BaseModel method, which is no longer supported, and replacing it with model_dump_json().
Fixed issues #1304 and #1308 .

### CODE REVIEW

File: tools/generate_dfvfs_test_data/generate_test_data_linux.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 #
 # Script to generate dfVFS test files on Linux.
-# Copied with permission from: https://github.com/log2timeline/dfvfs/blob/main/utils/generate_test_data_linux.sh
+# Copied with permission from: https://github.com/log2timeline/dfvfs/blob/master/main/utils/generate_test_data_linux.sh
```
Comment (jleaniz) [inline tools/generate_dfvfs_test_data/generate_test_data_linux.sh:4]: Are the changes to this file needed?

File: tools/generate_dfvfs_test_data/generate_test_data_macos.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 #
 # Script to generate dfVFS test files on Mac OS.
-# Copied with permission from: https://github.com/log2timeline/dfvfs/blob/main/utils/generate_test_data_macos.sh
+# Copied with permission from: https://github.com/log2timeline/dfvfs/blob/master/main/utils/generate_test_data_macos.sh
```
Comment (jleaniz) [inline tools/generate_dfvfs_test_data/generate_test_data_macos.sh:4]: Are the changes to this file needed?

File: tools/generate_dfvfs_test_data/generate_test_data_windows.bat [hunk 3]
```diff
@@ -1,7 +1,7 @@
 @echo off
 
 rem Script to generate dfVFS test files on Windows.
-rem Copied with permission from: https://github.com/log2timeline/dfvfs/blob/main/utils/generate_test_data_windows.bat
+rem Copied with permission from: https://github.com/log2timeline/dfvfs/blob/master/main/utils/generate_test_data_windows.bat
```
Comment (jleaniz) [inline tools/generate_dfvfs_test_data/generate_test_data_windows.bat:4]: Are the changes to this file needed?

File: turbinia/api/client/turbinia_api_lib/rest.py [hunk 4]
```diff
@@ -7,7 +7,6 @@
     Generated by: https://openapi-generator.tech
 """
 
-
```
Comment (jleaniz) [inline turbinia/api/client/turbinia_api_lib/rest.py:10]: Why do we need to modify this file in this pull request? Seems unrelated to the Pydantic method changes needed.

File: turbinia/api/routes/request.py [hunk 5]
```diff
@@ -47,10 +48,12 @@ async def get_requests_summary(request: Request):
     if not requests_summary.get_requests_summary():
       return JSONResponse(
           content={'detail': 'Request summary is empty'}, status_code=200)
-    response_json = requests_summary.json(sort_keys=True)
+    response_json = requests_summary.model_dump_json()
+    response = json.loads(response_json)
+    if isinstance(response, dict):
```
Comment (jleaniz) [inline turbinia/api/routes/request.py:53]: I think ```response_json``` will always be a dictionary in this scenario. Maybe try to catch ```JSONDecodeError```  from ```json.loads``` instead?

**Review by jleaniz — COMMENTED** [review]
I know this is still in draft but leaving some preliminary comments.

File: tools/generate_dfvfs_test_data/generate_test_data_linux.sh (hunk 1 above)
Comment (Igor8mr) [reply tools/generate_dfvfs_test_data/generate_test_data_linux.sh]: Reverted the changes from that file.

File: tools/generate_dfvfs_test_data/generate_test_data_macos.sh (hunk 2 above)
Comment (Igor8mr) [reply tools/generate_dfvfs_test_data/generate_test_data_macos.sh]: Reverted the changes from that file.

File: tools/generate_dfvfs_test_data/generate_test_data_windows.bat (hunk 3 above)
Comment (Igor8mr) [reply tools/generate_dfvfs_test_data/generate_test_data_windows.bat]: Reverted the changes to that file.

File: turbinia/api/client/turbinia_api_lib/rest.py (hunk 4 above)
Comment (Igor8mr) [reply turbinia/api/client/turbinia_api_lib/rest.py]: Reverted the changes to that file.

File: turbinia/api/routes/request.py (hunk 5 above)
Comment (Igor8mr) [reply turbinia/api/routes/request.py]: Thanks, I removed the if statement and added AttributeError to the exception block beneath it.
